src/d3a/models/market/blockchain_interface.py

```diff
@@ -59,11 +65,88 @@ def bc_listener(self):
         pass
 
 
+class SubstrateBlockchainInterface():
+    def __init__(self):
+        self.contracts = {'main': main_address}
+        self.substrate = SubstrateInterface(
+            url="ws://127.0.0.1:9944",
```

Comment:
Better to introduce and read from an environment variable than having a hard coded url here.

Reply:
Not sure where exactly to place it. Is it in the root, such as .env file?

Reply:
You can take [this](https://github.com/gridsingularity/d3a/blob/master/src/d3a/d3a_core/redis_connections/redis_communication.py#L35) as an example, so you define this url  default value at the top of module and eventually in production could be overwritten by respective url. Does it sounds like a good approach to you?

Reply:
Yes, sounds good, now it imports a default url from utils